docs/examples/plot_dynamic_beat.py

```diff
@@ -0,0 +1,132 @@
+       Journal of New Music Research 36.1 (2007): 51-60.
+       http://labrosa.ee.columbia.edu/projects/beattrack/
+"""
+
+# Code source: Brian McFee
+# License: ISC
+
+import numpy as np
+import librosa
+import matplotlib.pyplot as plt
+from IPython.display import Audio
+
+#######################################
+# ------------------------------
+# A recording with dynamic tempo
+# ------------------------------
+# The example recording included in this notebook consists of a drum beat that
+# gradually increases in tempo from 30 BPM to 240 BPM over a 30-second time interval.
+
+y, sr = librosa.load('audio/snare-accelerate.ogg')
+
+# %%
+# We can visualize the spectrogram of this recording and listen to it.
+# From the spectrogram, we can see that the spacing between drum beats becomes
+# smaller over time, indicating a gradual increase of tempo.
+
+fig, ax = plt.subplots()
+melspec = librosa.power_to_db(librosa.feature.melspectrogram(y=y, sr=sr), ref=np.max)
+librosa.display.specshow(melspec, y_axis='mel', x_axis='time', ax=ax)
+ax.set(title='Mel spectrogram')
+
+Audio(data=y, rate=sr)
+
+#########################################
+# --------------------------
+# Static tempo beat tracking
+# --------------------------
+# If we run the beat tracker on this signal directly and sonify the result as a click
+# track, we can see that it does not follow particularly well.
+# This is because the beat tracker is assuming a single (average) tempo across the
+# entire signal.
+# Note: here we are disabling trimming because we know the beats carry through to
+# the end of the signal.
+
+tempo, beats_static = librosa.beat.beat_track(y=y, sr=sr, units='time', trim=False)
+
+click_track = librosa.clicks(times=beats_static, sr=sr, click_freq=660,
+                             click_duration=0.25, length=len(y))
+
+print(f"Tempo estimate: {tempo[0]:.2f} BPM")
+Audio(data=y+click_track, rate=sr)
+
+#########################################
+# -------------
+# Dynamic tempo
+# -------------
+# Instead of using a single tempo, we can supply an array of tempo estimates
+# to the beat tracker.  This can be computed by disabling aggregation in the
+# tempo estimator as follows.
+
+tempo_dynamic = librosa.feature.tempo(y=y, sr=sr, aggregate=None, std_bpm=4)
+
+# %%
+# Here, we've expanded the standard deviation of the tempo estimator from its
+# default of 1 to 4.  This is to account for the broad range of tempo drift in this
+# particular recording (30 BPM to 240 BPM).
+# We can plot the dynamic tempo estimate, along with the static estimate, to see how
+# they differ.
+
+fig, ax = plt.subplots()
+times = librosa.times_like(tempo_dynamic, sr=sr)
+ax.plot(times, tempo_dynamic, label='Dynamic tempo estimate')
+ax.axhline(tempo, label='Static tempo estimate', color='r')
+ax.legend()
+ax.set(xlabel='Time [sec]', ylabel='Tempo [BPM]')
+
+# %%
+# The dynamic tempo estimate is computed (by default) using a 8-second sliding
```

Comment:
`librosa.feature.tempo` estimates tempo over a sliding window whose duration `ac_size` is equal to 8 seconds by default.
In this example, the result is not perfect: `tempo_dynamic` contains jagged and nonuniform steps. However, it does provide a rough picture of how tempo is changing over time.